packages/stencil/components/pagination/src/components/osds-pagination/osds-pagination.tsx

```diff
@@ -0,0 +1,200 @@
+import {
+  OdsLogger,
+  OdsPagination,
+  OdsPaginationController,
+  odsPaginationDefaultAttributes,
+  OdsPaginationEvents,
+  OdsPaginationMethods,
+  OdsPaginationCurrentChangeEventDetail,
+  OdsIconSize,
+  OdsIconName,
+  OdsButtonSize,
+  OdsButtonVariant,
+  OdsTextSize,
+} from '@ovhcloud/ods-core';
+import { HTMLStencilElement } from '@stencil/core/internal';
+import { OdsStencilEvents, OdsStencilMethods } from '@ovhcloud/ods-stencil/libraries/stencil-core';
+import { OdsThemeColorIntent } from '@ovhcloud/ods-theming';
+
+/**
+ * @slot (placeholder) - Pagination placeholder
+ */
+@Component({
+  tag: 'osds-pagination',
+  styleUrl: 'osds-pagination.scss',
+  shadow: true,
+})
+export class OsdsPagination implements OdsPagination<OdsStencilMethods<OdsPaginationMethods>, OdsStencilEvents<OdsPaginationEvents>> {
+  private logger = new OdsLogger('OsdsPagination');
+  controller: OdsPaginationController = new OdsPaginationController(this);
+
+  @Element() el!: HTMLStencilElement;
+
+  /** @see OdsPaginationAttributes.current */
+  @Prop({ reflect: true, mutable: true }) current: number = odsPaginationDefaultAttributes.current;
+
+  /** @see OdsPaginationAttributes.defaultCurrent */
+  @Prop({ reflect: true }) defaultCurrent?: number = odsPaginationDefaultAttributes.defaultCurrent;
+
+  /** @see OdsPaginationAttributes.totalPages */
+  @Prop({ reflect: true }) totalPages: number = odsPaginationDefaultAttributes.totalPages;
+
+  /** @see OdsPaginationAttributes.disabled */
+  @Prop({ reflect: true, mutable: true }) disabled: boolean = odsPaginationDefaultAttributes.disabled;
+
+  /** @see OdsPaginationEvents.odsCurrentChange */
+  @Event() odsCurrentChange!: EventEmitter<OdsPaginationCurrentChangeEventDetail>;
+
+  @Watch('defaultCurrent')
+  onDefaultCurrentChange(defaultCurrent?: number) {
+    this.logger.debug(`current: ${this.current}]`, 'defaultCurrent: ', defaultCurrent);
+  }
+
+  componentWillLoad() {
+    this.onDefaultCurrentChange();
+
+    if (this.defaultCurrent !== undefined) {
+      this.current = this.defaultCurrent;
+    }
+
+    this.current = this.current;
+  }
+
+  async componentDidUpdate() {
+    (this.el.shadowRoot?.querySelector('.selectedpage') as HTMLStencilElement)?.focus();
+  }
+
+  @Watch('current')
+  async onCurrentChange(current: number, oldCurrent?: number) {
+    this.logger.log(`current: ${this.current}]`, 'current changed. emit new current', { current });
+    this.emitChange(current, oldCurrent);
+  }
+
+  private emitChange(current: number, oldCurrent?: number) {
+    this.logger.debug('emit', { current, oldCurrent });
+    this.odsCurrentChange.emit({
+      current: current,
+      oldCurrent: oldCurrent,
+    });
+  }
+
+  /**
+   * @internal
+   * @see OdsPaginationMethods.setPageIndex
+   */
+  @Method()
+  async setPageIndex(current: number) {
+    this.current = current;
+  }
+
+  onKeyPress = (event: any, page: any) => {
+    if (event.keyCode === 13 || event.keyCode === 32) {
```

Comment:
this logic has to be moved inside the controller like this:
```typescript jsx
// component
onKeyPress(event: KeyboardEvent, page: number) {
      this.controller.handleKeyPress(event, page);
  }

// controller
onKeyPress(event: KeyboardEvent, page: number) {
  if (event.keyCode === 13 || event.keyCode === 32) {
    this.controller.setPageIndex(page);
  }
}
```

(move the max logic into controller)